relation_processor.py

```diff
@@ -0,0 +1,181 @@
+# Process data and create dataloader for models
+# For binary relation extraction
+
+import pandas as pd
+from collections import Counter
+from sklearn.model_selection import train_test_split
+from torch import LongTensor
+from torch.nn.utils.rnn import pad_sequence
+from torch.utils.data import TensorDataset, DataLoader
+
+
+class RelationProcessor:
+    """Data Processor for relation extraction task
+
+    Process data and covert text into numbers 
+    Create train/val/test DataLoader for models usage
+
+    Attributes:
+        data: Data in correct format
+        batch_size: Size of each batch being loaded from DataLoader
+        val_test_ratio: Total ratio of data is split into val and test set.
+            Example: val_test_ratio = 0.3 means val_ratio = test_ration = 0.15
+        vocab: Dict of unique tokens (in given data) mapped to numbers
+
+    """
+
+    def __init__(self, data: list[dict], batch_size: int, val_test_ratio: float):
+        """Initialize RelationProcessor 
+
+        Args:
+            data: Data in correct format
+            batch_size: Size of each batch being loaded from DataLoader
+            val_test_ratio: Total ratio of data is split into val and test set.
+                Example: val_test_ratio = 0.3 means val_ratio = test_ration = 0.15
+
+        """
+    
+        self.data = data
+        self.val_test_ratio = val_test_ratio
+        self.batch_size = batch_size
+        self.vocab = None
+
+    def run(self) -> tuple[DataLoader, DataLoader, DataLoader, LongTensor]:
+        """Run RelationProcessor 
+
+        Create vocab out of given data, add typed markers
+        Create train/val/test DataLoader for models usage
+
+        Returns:
+            DataLoader for train dataset
+            DataLoader for val dataset
+            DataLoader for test dataset
+            Category labels for test dataset
+
+        """
+
+        self.vocab = self._create_vocab()
+
+        self._add_typed_markers()
+        
+        sequences, labels = self._create_sequences_and_labels()
+
+        train_loader, val_loader, test_loader, y_test = self._create_dataset(sequences, labels)
+
+        return train_loader, val_loader, test_loader, y_test
+
+    def get_vocab_size(self) -> int:
+        """Calculate size of vocabulary 
+
+        Returns:
+            Size of vocabulary  
+
+        """
+
+        return len(self.vocab)
+
+    def _create_vocab(self) -> dict:
+        word_counter = Counter()
+
+        for item in self.data:
+            word_counter.update(item['sentence'])
+        
+        vocabulary = {word: i+1 for i, (word, _) in enumerate(word_counter.most_common())}
+        vocabulary['<UNK>'] = 0
+        vocabulary['<PAD>'] = len(vocabulary)
+
+        return vocabulary
+
+    def _add_typed_markers(self) -> None:
```

Comment:
In a future PR we will clean up this code, since the function is long, but for right now it works